spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/repository/query/PartTreeFirestoreQuery.java

```diff
@@ -88,18 +88,31 @@ public PartTreeFirestoreQuery(FirestoreQueryMethod queryMethod, FirestoreReactiv
 		this.tree = new PartTree(queryMethod.getName(), returnedType.getDomainType());
 		this.persistentEntity = mappingContext.getPersistentEntity(returnedType.getDomainType());
 		this.classMapper = classMapper;
+		validate();
 	}
 
-	@Override
-	public Object execute(Object[] parameters) {
+	private void validate() {
 		List parts = this.tree.get().collect(Collectors.toList());
 		if (parts.size() > 1 && parts.get(0) instanceof PartTree.OrPart) {
 				throw new FirestoreDataException(
-						"Cloud Firestore only supports multiple filters combined with AND.");
+						"Cloud Firestore doesn't support 'OR' (" + this.getQueryMethod().getName() + ")");
```

Comment:
Would it help the caller to see (method name: [METHOD]) as opposed to just ([METHOD])?